libavcodec/libaribcaption.c

```diff
@@ -0,0 +1,1141 @@
+                av_bprintf(&buf, "{\\fscx%"PRId64"}",
+                           av_rescale(ch->char_width, 100, ctx->font_size));
+                char_width = ch->char_width;
+            }
+            if (ch->char_height != char_height) {
+                av_bprintf(&buf, "{\\fscy%"PRId64"}",
+                           av_rescale(ch->char_height, 100, ctx->font_size));
+                char_height = ch->char_height;
+            }
+            if (ch->style != charstyle) {
+                aribcc_charstyle_t diff = ch->style ^ charstyle;
+                if (diff & ARIBCC_CHARSTYLE_STROKE) {
+                    if (charstyle & ARIBCC_CHARSTYLE_STROKE) {
+                        if (ctx->force_stroke_text)
+                            av_bprintf(&buf, "{\\bord%d}",
+                                       (int)(ctx->stroke_width * 4.0 / 3.0));
+                        else
+                            av_bprintf(&buf, "{\\bord0}");
+                    } else
+                        av_bprintf(&buf, "{\\bord3}");
+                }
+                if (diff & ARIBCC_CHARSTYLE_BOLD) {
+                    if (charstyle & ARIBCC_CHARSTYLE_BOLD)
+                        av_bprintf(&buf, "{\\b0}");
+                    else
+                        av_bprintf(&buf, "{\\b1}");
+                }
+                if (diff & ARIBCC_CHARSTYLE_ITALIC) {
+                    if (charstyle & ARIBCC_CHARSTYLE_ITALIC)
+                        av_bprintf(&buf, "{\\i0}");
+                    else
+                        av_bprintf(&buf, "{\\i1}");
+                }
+                if (diff & ARIBCC_CHARSTYLE_UNDERLINE) {
+                    if (charstyle & ARIBCC_CHARSTYLE_UNDERLINE)
+                        av_bprintf(&buf, "{\\u0}");
+                    else
+                        av_bprintf(&buf, "{\\u1}");
+                }
+                charstyle = ch->style;
+            }
+            if (ch->text_color != text_color) {
+                set_ass_color(&buf, 1, ch->text_color, text_color);
+                text_color = ch->text_color;
+            }
+            if (ch->stroke_color != stroke_color) {
+                set_ass_color(&buf, 3, ch->stroke_color, stroke_color);
+                stroke_color = ch->stroke_color;
+            }
+            if (ch->back_color != back_color) {
+                if (ctx->border_style == 4)
+                    set_ass_color(&buf, 4, ch->back_color, back_color);
+                else
+                    set_ass_color(&buf, 3, ch->back_color, back_color);
+                back_color = ch->back_color;
+            }
+            if (region->chars[j].type == ARIBCC_CHARTYPE_DRCS)
+                av_bprintf(&buf, "\xe3\x80\x93");  /* Geta Mark */
+            else
+                ff_ass_bprint_text_event(&buf, ch->u8str, strlen(ch->u8str), "", 0);
+        }
+        av_bprintf(&buf, "{\\r}%s",
+                         (i + 1 < ctx->caption.region_count) ? "\\N" : "");
```

Comment:
This line only makes sense on the `ass_workaround` path, and the `{\\r}` is only necessary if inserting the newline, so this could be:
```suggestion
        if (ctx->ass_workaround && (i + 1 < ctx->caption.region_count))
                av_bprintf(&buf, "{\\r}\\N");
```

Reply:
The `{\\r}` is diverted from former `libaribb24`.
It seems it is not required for multi-rect ASS, I will change the code as you suggested.